src/agent.cpp

```diff
@@ -202,8 +208,9 @@ JNIEXPORT jobjectArray JNICALL Java_com_intellij_memory_agent_proxy_IdeaNativeAg
         jclass thisClass,
         jobject startObject,
         jobjectArray suspectClass,
-        jlong timeoutInMillis) {
-    return GetFirstReachableObjectOfClassAction(env, gdata->jvmti).runWithTimeout(timeoutInMillis, startObject, suspectClass);
+        jlong timeoutInMillis,
+        jstring cancellationFileName) {
+    return GetFirstReachableObjectOfClassAction(env, gdata->jvmti).runWithTimeout(timeoutInMillis, cancellationFileName, startObject, suspectClass);
```

Comment:
Adding more parameters is becoming too hard :( A possible solution - replace the static methods with instance ones and configure the common arguments by passing to a constructor from Java:

```
MemoryAgentProxy(timeout, cancellationFile).getRetainedSizeByClasses(classesArray);
```

WDYT?

Reply:
The idea is nice, I will implement it